pinot-common/src/main/java/com/linkedin/pinot/common/utils/ServiceStatus.java

```diff
@@ -123,147 +128,130 @@ public String getStatusDescription() {
-    protected String _instanceName;
-    protected List<String> _resourcesToMonitor;
-    protected boolean _finishedStartingUp = false;
-    protected HelixManager _helixManager;
-    protected HelixAdmin _helixAdmin;
-    protected HelixDataAccessor _helixDataAccessor;
-    protected Builder _keyBuilder;
-    protected Set<String> _resourcesDoneStartingUp = new HashSet<>();
+    protected final String _clusterName;
+    protected final String _instanceName;
+    protected final HelixAdmin _helixAdmin;
+    protected final HelixDataAccessor _helixDataAccessor;
+
+    private final Set<String> _resourcesToMonitor;
+    private final int _numTotalResourcesToMonitor;
+
     private String _statusDescription = STATUS_DESCRIPTION_INIT;
 
     public IdealStateMatchServiceStatusCallback(HelixManager helixManager, String clusterName, String instanceName) {
-      _helixManager = helixManager;
       _clusterName = clusterName;
       _instanceName = instanceName;
+      _helixAdmin = helixManager.getClusterManagmentTool();
+      _helixDataAccessor = helixManager.getHelixDataAccessor();
 
-      if (_helixManager != null) {
-        _helixAdmin = _helixManager.getClusterManagmentTool();
-        _helixDataAccessor = _helixManager.getHelixDataAccessor();
-        _keyBuilder = _helixDataAccessor.keyBuilder();
-      }
-
-      // Make a list of the resources to monitor
-      _resourcesToMonitor = new ArrayList<>();
-
-      for (String resource : _helixAdmin.getResourcesInCluster(_clusterName)) {
+      _resourcesToMonitor = new HashSet<>();
+      for (String resourceName : _helixAdmin.getResourcesInCluster(_clusterName)) {
         // Only monitor table resources and broker resource
-        if (!TableNameBuilder.isTableResource(resource) && !resource.equals(
+        if (!TableNameBuilder.isTableResource(resourceName) && !resourceName.equals(
             CommonConstants.Helix.BROKER_RESOURCE_INSTANCE)) {
           continue;
         }
-        final IdealState idealState = getResourceIdealState(resource);
-        for (String partitionInResource : idealState.getPartitionSet()) {
-          if (idealState.getInstanceSet(partitionInResource).contains(_instanceName)) {
-            _resourcesToMonitor.add(resource);
-            break;
+        // Only monitor enabled resources
+        IdealState idealState = getResourceIdealState(resourceName);
+        if (idealState.isEnabled()) {
+          for (String partitionName : idealState.getPartitionSet()) {
+            if (idealState.getInstanceSet(partitionName).contains(_instanceName)) {
+              _resourcesToMonitor.add(resourceName);
+              break;
+            }
           }
         }
       }
+      _numTotalResourcesToMonitor = _resourcesToMonitor.size();
 
-      LOGGER.info("Monitoring resources {} for start up of instance {}", _resourcesToMonitor, _instanceName);
+      LOGGER.info("Monitoring {} resources: {} for start up of instance {}", _numTotalResourcesToMonitor,
```

Comment:
Logging of complete resource list was omitted from this log message to save the length of the log line. You may want to log a truncated version of the list/set or not log it at all (i.e. leave the log line as is)

Reply:
The current one lists the complete list. This one just logs the number of resources as well.